cli/src/cli.rs

```diff
@@ -1178,17 +1238,27 @@ fn process_deploy(
         config.commitment,
     )?;
 
-    trace!("Creating program account");
-    let result = rpc_client.send_and_confirm_transaction_with_spinner_and_config(
-        &create_account_tx,
-        config.commitment,
-        config.send_transaction_config,
-    );
-    log_instruction_custom_error::<SystemError>(result, &config).map_err(|_| {
-        CliError::DynamicProgramError("Program account allocation failed".to_string())
-    })?;
+    if let Some(message) = initial_message {
+        trace!("Creating or modifying program account");
+        let num_required_signatures = message.header.num_required_signatures;
 
-    let (blockhash, _, _) = rpc_client
+        let mut initial_transaction = Transaction::new_unsigned(message);
+        if num_required_signatures == 2 {
```

Comment:
It's not immediately clear to me what this logic is for.  Maybe a comment is warranted?